src/core/regimes.rkt

```diff
@@ -33,23 +33,19 @@
     (cond
      [(null? alts) '()]
      [else
-      (define recomputed-branch-exprs
-        (if (flag-set? 'reduce 'branch-expressions)
-            (exprs-to-branch-on alts ctx)
-            (context-vars ctx)))
-      
-      (define recomp-errs 
-        (for/hash ([bexpr recomputed-branch-exprs]) (values bexpr (hash-ref errs bexpr))))
-
       (define-values (opt new-errs) 
-        (infer-splitpoints alts recomputed-branch-exprs recomp-errs ctx))
+        (infer-splitpoints alts errs ctx))
```

Comment:
The idea here is that `recomp-errs` isn't used? Or that it's OK if it's a superset?

Reply:
I was recreating the hash table with only the recomputed branch expressions but we only iterated through the values of the recomputed branch expressions, not the keys of the hash table, so there wasn't actually any need to do that